x-pack/test/api_integration/deployment_agnostic/apis/console/spec_definitions.ts

```diff
@@ -6,28 +6,25 @@
  */
 
 import expect from '@kbn/expect';
-import { RoleCredentials, InternalRequestHeader } from '@kbn/ftr-common-functional-services';
 import { DeploymentAgnosticFtrProviderContext } from '../../ftr_provider_context';
+import { SupertestWithRoleScopeType } from '../../services';
 
 export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
-  const samlAuth = getService('samlAuth');
-  const supertestWithoutAuth = getService('supertestWithoutAuth');
-  let roleAuthc: RoleCredentials;
-  let internalHeaders: InternalRequestHeader;
+  const roleScopedSupertest = getService('roleScopedSupertest');
+  let supertestWithAdminScope: SupertestWithRoleScopeType;
 
   describe('GET /api/console/api_server', () => {
     before(async () => {
-      roleAuthc = await samlAuth.createM2mApiKeyWithRoleScope('admin');
-      internalHeaders = samlAuth.getInternalRequestHeader();
+      supertestWithAdminScope = await roleScopedSupertest.getSupertestWithRoleScope('admin', {
+        withInternalHeaders: true,
+      });
     });
     after(async () => {
-      await samlAuth.invalidateM2mApiKeyWithRoleScope(roleAuthc);
+      await supertestWithAdminScope.destroy();
     });
     it('returns autocomplete definitions', async () => {
-      const { body } = await supertestWithoutAuth
+      const { body } = await supertestWithAdminScope
         .get('/api/console/api_server')
-        .set(roleAuthc.apiKeyHeader)
-        .set(internalHeaders)
         .set('kbn-xsrf', 'true')
```

Comment:
Do you think there's any utility in managing this header too?

Reply:
yes, and actually I added it.

```
const supertestWithAdminScope = await roleScopedSupertest.getSupertestWithRoleScope('admin', {
  withInternalHeaders: true,
  withCustomHeaders: {'kbn-xsrf': 'true'},
});
```

I probably forgot to update this file, will push an update.

Reply:
9acb49f206405c20ad83b28105ac55b60f817ca2